### ISSUE TITLE
Unspecified GSS failure.

### ISSUE THREAD

**Issue by zckrs** [dmansfield/passport-negotiate#2]
Hi,

I try to use passport-negotiate to replace our actual authentication method written in JAVA.

But I have a strange error. Maybe can you help me.

`authentication failed at operation 'init' with error: Error: authenticate_gss_server_init(gss_acquire_cred): Unspecified GSS failure.  Minor code may provide more information:`

I run our ExpressJS server with KRB5_KTNAME=/data/myAwesome.keytab
I have the REALM define in /etc/krb5.conf

What I have misunderstood ?

**Comment by dmansfield** [comment]
I have few questions:

1) What is the output of `klist -k /data/myAwesome.keytab`

2) What are you passing in as options when the strategy is created?  In particular, are you passing anything in as the options.servicePrincipalName?  If not, try passing in one which matches one of the keys in your keytab (see question 1) except it should be formatted as HTTP@my.host.com instead of HTTP/my.host.com

3) There should be additional text after the "Minor code may provide more information" which should be useful, can you check?

**Comment by zckrs** [comment]
1) `klist` is not installed on your machine

2) 

```
passport.use('kerberos', new NegotiateStrategy({
    servicePrincipalName: 'HTTP/sacch15520.ad.sys@AD.SYS',
    verbose: true
  }, function (principal, done) {
    console.log('***************');
    console.log(principal);
    console.log(arguments);
    done(null, principal);
  }));
```

3) I give you all text returned by cli :-( 

EDIT: just trying with HTTP@sacch15520.ad.sys@AD.SYS ==> same result

**Comment by dmansfield** [comment]
Ok.  Well without klist it's hard to verify that a key for your principal is present in the keytab.  But let's move on.  Check the permissions on the keytab, make sure the file is readable by the node process, and that selinux isn't preventing access (are you running on linux??)

Can you try removing the realm (@AD.SYS) from the servicePrincipalName in the options?

Can you check that the hostname is set to "sacch15520.ad.sys" and that forward and reverse DNS for this hostname is working?

Also, are you sure you exported the KRB5_KTNAME environment variable or set in internally to the node process?

Just some random thoughts.

**Comment by zckrs** [comment]
First thank you for your interest :D. FYI since 2 months I try to work with that.
- OS: CentOS 6
- keytab mod: 777
- Without REALM ===> `/*************/node_modules/passport-negotiate/lib/passport-negotiate/strategy.js:111 req.session.authenticatedPrincipal = principal; TypeError: Cannot set property 'authenticatedPrincipal' of undefined`
- I am connected on sacch15520.ad.sys
- I run my Express server with `KRB5_KTNAME=/data/myAwesome.keytab  node app.js`

**Comment by dmansfield** [comment]
The error on line 111 is progress, believe it or not!  Authentication has succeeded, however you don't have a session established and the code expects to always have a session.  I fixed that in September in my branch here...  I haven't released a new version in some time I guess.

So maybe you can change that to 

```
if (req.session) req.session.authenticatedPrincipal = principal;
```

**Comment by dmansfield** [comment]
Latest code has been released as 0.1.4 and should fix the issue.
